geoh5py/shared/entity.py

```diff
@@ -310,15 +310,22 @@ def metadata(self, value: dict | np.ndarray | bytes | None):
         if self.on_file:
             self.workspace.update_attribute(self, "metadata")
 
-    def update_metadata(self, value: dict | None):
+    def update_metadata(self, value: dict):
```

Comment:
```suggestion
    def update_metadata(self, value: dict | None):
```
The None allows to clear out metadata.